Source/Scene/Expression.js

```diff
@@ -960,7 +960,13 @@ define([
     // that we can assign if we know the types before runtime
 
     Node.prototype._evaluateNot = function(frameState, feature) {
-        return !(this._left.evaluate(frameState, feature));
+        var left = this._left.evaluate(frameState, feature);
+        //>>includeStart('debug', pragmas.debug);
+        if (typeof(left) !== 'boolean') {
+            throw new DeveloperError('Error: Operation is undefined.');
```

Comment:
These exceptions are all fine, but can we be more precise about the type mismatch and any other info we can provide to help devs track down the issue?